t/Net/DRI/Protocol/EPP/Extensions/JP.t

```diff
@@ -0,0 +1,357 @@
+
+
+
+####################################################################################################
+### Domain create - organizational JPEX TLDs
+####################################################################################################
+# let's use the error create response from their documentation to add some extra tests/echecks
+$R2='';
+$cs=$dri->local_object('contactset');
+$c1=$dri->local_object('contact')->srid('sh8011');
+$c2=$dri->local_object('contact')->srid('sh8012');
+$c3=$dri->local_object('contact')->srid('sh8013');
+$cs->set($c1,'registrant');
+$cs->set($c2,'admin');
+$cs->set($c3,'tech');
+throws_ok { $dri->domain_create('test.co.jp',{pure_create=>1,duration=>DateTime::Duration->new(years=>2),contact=>$cs,auth=>{pw=>'2fooBAR'}}) } qr/Invalid duration/, 'domain create (ojp) for 2 years invalid - only 1 year accepted';
+# note("missing required suffix");
+throws_ok { $dri->domain_create('test.co.jp',{pure_create=>1,duration=>DateTime::Duration->new(years=>1),contact=>$cs,auth=>{pw=>'2fooBAR'}}) } qr/suffix is required/, 'domain create (ojp) mandatory suffix is missing';
+# note("invalid suffix");
+throws_ok { $dri->domain_create('test.co.jp',{pure_create=>1,duration=>DateTime::Duration->new(years=>1),contact=>$cs,auth=>{pw=>'2fooBAR'}, suffix=>'jpp'}) } qr/invalid suffi/, 'domain create (ojp) invalid suffix usage';
+# note("invalid domainCreatePreValidation");
+throws_ok { $dri->domain_create('test.co.jp',{pure_create=>1,duration=>DateTime::Duration->new(years=>1),contact=>$cs,auth=>{pw=>'2fooBAR'}, suffix=>'ojp', domainCreatePreValidation=>'Y'}) } qr/invalid domainCreatePreValidation/, 'domain create (ojp) invalid domainCreatePreValidation usage';
+my %organization = (
+	'name' => '株式会社 ドメインサンプル',
+	'nameKana' => 'かぶしきかいしゃどめいんさんぷる',
+	'nameEn' => 'Domain-sample Corporatin',
+	'postalCode' => '101-0065',
+	'address' => '東京都千代田区西神田 11-22-33',
+	'addressEn' => '11-22-33 Nishi-Kanda Chiyoda-ku, Tokyo Japan, 101-0065',
+	'type' => '株式会社',
+	'typeEn' => 'Company Limited',
+	'registrationDate' => '2000-04-03',
+	'registrationAddress' => '東京都千代田区西神田 11-22-33',
+	'replyMail' => 'example@example.shop'
+);
+my %representative = (
+	'lastName' => 'ドメイン',
+	'firstName' => '太郎',
+	'lastNameEn' => 'Domain',
+	'firstNameEn' => 'Taro',
+	'title' => '代表取締役'
+);
+my %condition = (
+	'no_other_ojp_domain_is_registered_under_the_same_corporation' => 'yes',
+	'the_applicant_must_a_registered_corporation' => 'yes',
```

Comment:
A re my earlier comment, I think I see what you were doing - you were checking if one of these two conditions were set ...?

Reply:
Yes. Plus, it really needs to be `uc` (https://github.com/thedarkwinter/Net-DRI/pull/432/files#diff-2a4a24d6e46f8ea7603cd2e91cea17d97de7442d4b3c879461b0ce2aeb887a42R339) otherwise error from server side.